src/test/java/duke/logic/EditorManagerTest.java

```diff
@@ -0,0 +1,101 @@
+        EventList events = new EventList();
+        events.add(e);
+        venues.add(e.getLocation());
+        EditorManager.activate(events, venues);
+    }
+
+    @Test
+    void deactivate() {
+        EditorManager.deactivate();
+        assertFalse(EditorManager.isActive());
+    }
+
+    @Test
+    void isActive() throws DukeException {
+        EditorManager.deactivate();
+        assertFalse(EditorManager.isActive());
+        Event e = new Event("Pulau Tekong", LocalDateTime.now(), LocalDateTime.now());
+        VenueList venues = new VenueList();
+        EventList events = new EventList();
+        events.add(e);
+        venues.add(e.getLocation());
+        EditorManager.activate(events, venues);
+        assertTrue(EditorManager.isActive());
+    }
+
+    @Test
+    void edit() throws DukeException {
+        EditorManager.deactivate();
+        Event e = new Event("Pulau Tekong", LocalDateTime.now(), LocalDateTime.now());
+        VenueList venues = new VenueList();
+        EventList events = new EventList();
+        events.add(e);
+        venues.add(e.getLocation());
+        Event e2 = new Event("Tuas", LocalDateTime.now(), LocalDateTime.now());
+        events.add(e2);
+        venues.add(e2.getLocation());
+        EditorManager.activate(events, venues);
+        assertTrue(EditorManager.isActive());
+        assertThrows(EventNotSelectedException.class, () -> EditorManager.edit("Yew Tee"));
+        EditorManager.edit(KeyCode.ENTER);
+        assertTrue(EditorManager.edit("Yew Tee") instanceof PromptCommand);
+        EditorManager.edit(KeyCode.DOWN);
+        assertTrue(EditorManager.edit("Mon") instanceof PromptCommand);
+        EditorManager.edit(KeyCode.DOWN);
+        assertTrue(EditorManager.edit("Mon") instanceof PromptCommand);
+        EditorManager.edit(KeyCode.DOWN);
+        assertTrue(EditorManager.edit("Boon lay") instanceof PromptCommand);
+        assertTrue(EditorManager.edit("x") instanceof EditCommand);
+    }
+
+    @Test
+    void testEdit() throws DukeException {
+        EditorManager.deactivate();
+        assertFalse(EditorManager.isActive());
+        Event e = new Event("Pulau Tekong", LocalDateTime.now(), LocalDateTime.now());
+        VenueList venues = new VenueList();
+        EventList events = new EventList();
+        events.add(e);
+        venues.add(e.getLocation());
+        EditorManager.activate(events, venues);
+        PanelResult result = EditorManager.edit(KeyCode.K);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.UP);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ENTER);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ENTER);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ESCAPE);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ESCAPE);
+        assertTrue(result.isReady());
```

Comment:
Is it possible to do negative testing?

Reply:
I don't think so, for this particular method, I'm testing how the EditorManager will handle keyEvents, and under no circumstances the Class should fail, according to the design.